src/Apache.IoTDB/DataStructure/SessionDataSet.cs

```diff
@@ -32,284 +32,104 @@ public class SessionDataSet : System.IDisposable
-                                break;
-                            case TSDataType.DOUBLE:
-                                localField = columnValueBuffer.GetDouble();
-                                break;
-                            case TSDataType.TEXT:
-                            case TSDataType.STRING:
-                                // case TSDataType.BLOB:
-                                localField = columnValueBuffer.GetStr();
-                                break;
-                            case TSDataType.BLOB:
-                                localField = columnValueBuffer.GetBinary();
-                                break;
-                            // TODO
-                            default:
-                                string err_msg = "value format not supported";
-                                throw new TException(err_msg, null);
-                        }
+        public DateTime GetTimestampByIndex(int columnIndex) => _rpcDataSet.GetTimestampByIndex(columnIndex);
+        public DateTime GetTimestamp(string columnName) => _rpcDataSet.GetTimestamp(columnName);
 
-                        fieldList.Add(localField);
-                    }
-                    else
-                    {
-                        localField = null;
-                        fieldList.Add(DBNull.Value);
-                    }
-                }
-            }
+        public DateTime GetDateByIndex(int columnIndex) => _rpcDataSet.GetDateByIndex(columnIndex);
+        public DateTime GetDate(string columnName) => _rpcDataSet.GetDate(columnName);
 
-            long timestamp = _timeBuffer.GetLong();
-            _rowIndex += 1;
-            _cachedRowRecord = new RowRecord(timestamp, fieldList, _columnNames);
-        }
+        public Binary GetBlobByIndex(int columnIndex) => _rpcDataSet.GetBinaryByIndex(columnIndex);
+        public Binary GetBlob(string columnName) => _rpcDataSet.GetBinary(columnName);
 
-        private bool IsNull(int loc, int row_index)
-        {
-            byte bitmap = _currentBitmap[loc];
-            int shift = row_index % 8;
-            return ((Flag >> shift) & bitmap) == 0;
-        }
+        public int FindColumn(string columnName) => _rpcDataSet.FindColumn(columnName);
 
-        private bool FetchResults()
-        {
-            _rowIndex = 0;
-            var req = new TSFetchResultsReq(_client.SessionId, _sql, FetchSize, _queryId, true)
-            {
-                Timeout = DefaultTimeout
-            };
-            try
-            {
-                var task = _client.ServiceClient.fetchResultsAsync(req);
+        public IReadOnlyList<string> GetColumnNames() => _rpcDataSet._columnNameList;
+        public IReadOnlyList<string> GetColumnTypes() => _rpcDataSet._columnTypeList;
 
-                var resp = task.ConfigureAwait(false).GetAwaiter().GetResult();
+        public RowRecord GetRow() => _rpcDataSet.GetRow();
```

Comment:
Let's keep the user's performance unchanged
`HasNext`
`Next`
This can be used as a new method, but the semantics of the old method should not be changed
`public RowRecord GetRow() => _rpcDataSet.GetRow();`

Reply:
Resolved. All interfaces remain unchanged from the previous version